testing/ngr.py

```diff
@@ -0,0 +1,226 @@
+"""
+-----
+About
+-----
+
+Next Generation Runner (ngr): Effortless invoke programs and test harnesses.
+
+-------
+Backlog
+-------
+- After a few iterations, refactor to separate package.
```

Comment:
`ngr.py` will only be incubated here, it will be refactored to a standalone package after a few iterations.

Reply:
`testing/ngr.py` moved into `pueblo.ngr` already, and learned three more directory shapes to invoke: `javascript`, `make`, and `rust`.

```python
class ItemType(Enum):
    DOTNET = "dotnet"
    JAVA = "java"
    JAVASCRIPT = "javascript"
    MAKE = "make"
    PHP = "php"
    PYTHON = "python"
    RUBY = "ruby"
    RUST = "rust"
```

- https://github.com/pyveci/pueblo/pull/6/commits/b7a25308ace20